app/src/main/kotlin/com/ayatk/biblio/di/ViewModelFactory.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016-2018 ayatk.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ayatk.biblio.di
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+import javax.inject.Inject
+import javax.inject.Provider
+
+class ViewModelFactory @Inject constructor(
+    private val creators: Map<Class<out ViewModel>, @JvmSuppressWildcards Provider<ViewModel>>
+) : ViewModelProvider.Factory {
+
+  override fun <T : ViewModel> create(modelClass: Class<T>): T {
+    var creator: Provider<out ViewModel>? = creators[modelClass]
+    if (creator == null) {
+      for ((key, value) in creators) {
+        if (modelClass.isAssignableFrom(key)) {
+          creator = value
+          break
+        }
+      }
+    }
+    if (creator == null) {
+      throw IllegalArgumentException("unknown model class " + modelClass)
+    }
+    try {
+      @Suppress("UNCHECKED_CAST")
+      return creator.get() as T
+    } catch (e: Exception) {
```

Comment:
**[detekt]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Thrown exception is too generic. Prefer throwing project specific exceptions to handle error cases.